Sources/PubNub/Subscription/SubscriptionSession.swift

```diff
@@ -83,12 +132,33 @@ public class SubscriptionSession {
     at cursor: SubscribeCursor? = nil,
     withPresence: Bool = false
   ) {
-    strategy.subscribe(
-      to: channels,
-      and: groups,
-      at: cursor,
-      withPresence: withPresence
+    let channelSubscriptions = channels.map {
+      channel($0).subscription(
+        queue: queue,
+        options: withPresence ? ReceivePresenceEvents() : SubscriptionOptions.empty()
+      )
+    }
+    let channelGroupSubscriptions = groups.map {
+      channelGroup($0).subscription(
+        queue: queue,
+        options: withPresence ? ReceivePresenceEvents() : SubscriptionOptions.empty()
+      )
+    }
+    internalSubscribe(
+      with: channelSubscriptions,
+      and: channelGroupSubscriptions,
+      at: cursor?.timetoken
     )
+    channelSubscriptions.forEach { subscription in
+      subscription.subscriptionNames.flatMap { $0 }.forEach {
+        globalChannelSubscriptions[$0] = subscription
+      }
+    }
+    channelGroupSubscriptions.forEach { subscription in
+      subscription.subscriptionNames.flatMap { $0 }.forEach {
+        globalGroupSubscriptions[$0] = subscription
+      }
+    }
```

Comment:
As I understand, this one is for PubNub client instance itself? 
If subscribing to multiple entities, then it will be more reasonable to create `SubscriptionSet` with `Subscriptions`.

Reply:
I ran it by the team. You're right that this should be kept in `SubscriptionSet` under the hood. Unfortunately, Swift SDK is the only one that provides functionality like unsubscribing from Presence channels only. And this is what prevents us from using `SubscriptionSet` for global subscriptions. We have no API/method for SubscriptionSet to unsubscribe from Presence channel only. I could deprecate [this method](https://github.com/pubnub/swift/blob/master/Sources/PubNub/PubNub.swift#L399) or even remove `withPresence` parameter, but this decision requires PM input. Can we leave it as it is and fix it in case of further decisions?